### ISSUE TITLE
MyPy links for __init__.py find the wrong file

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#3181]
See PR #3215.

**Status**

- The crash fixed via #3183.
  This PR has been merged into the `devel` and `6.7.2` branches.
- The rest of this issue must wait until Leo 6.7.3. It might be fairly easy to do.
  - To do: add a new unit test.

**Original Report**

I ran the mypy command against an external file.  I got a mypy log pane message:

```
mypy: lst_sqr_extrap.py
Node: @file lst_sqr_extrap.py:638: error: Positional-only parameters are only supported in Python 3.8 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
mypy: finished
```

However, that message applied to the numpy import, not to my program.  This is shown in the console message:

```
C:\Users\tom\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy\__init__.pyi:638: error: Positional-only parameters are only supported in Python 3.8 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

A link to the error was displayed in the log pane, except that the link tried to go to the reported line in my external file instead of numpy.

In addition, when I clicked on the link, it vanished from the log pane and this message was printed to the console:

```
Uncaught exception in Leo...
Traceback (most recent call last):
  File "c:\Tom\git\leo-editor\leo\plugins\qt_frame.py", line 2705, in linkClicked
    g.handleUrl(url, c=self.c)
  File "c:\Tom\git\leo-editor\leo\core\leoGlobals.py", line 7388, in handleUrl
    return g.handleUnl(url, c)
  File "c:\Tom\git\leo-editor\leo\core\leoGlobals.py", line 7365, in handleUnl
    p = g.findUNL(unl.split("-->"), c2) or c2.rootPosition()
  File "c:\Tom\git\leo-editor\leo\core\leoGlobals.py", line 7245, in findUNL
    g.trace(f"Not found: global line {n} in {p.h}")
AttributeError: 'NoneType' object has no attribute 'h'
```

**Comment by edreamleo** [comment]
@tbpassin Thanks for this report. I have known about the problem with `__init__.py` files for awhile. It would be tricky to fix.

I'll fix the crash for Leo 6.7.2.

**Comment by tbpassin** [comment]
To avoid an if...else... style of guard, this kind of expression guards against the problem and I think its pretty readable:

`'5' in (p and p.h or '')`

Of course, it requires p.h to exist and be a string if p is not None, but I'd say that's a given in this use.

An alternative is

`'5' in (p.h if p else '')`

For some reason I like the first form better.

**Comment by edreamleo** [comment]
@tbpassin It took me awhile to understand your comment because it doesn't apply to the PR.

The original code was:
```python
p, offset, ok = c.gotoCommands.find_file_line(-n, p)  # Calls c.redraw().
if not ok:
    g.trace(f"Not found: global line {n} in {p.h}")
```
The new code changes the last line to
```python
g.trace(f"Not found: global line {n}")
```
No guard, of any form, will do any good here because `p` is *always* `None` if `ok` is `False`, a you can verify by looking at `find_file_line`.

**Comment by tbpassin** [comment]
It's true, I was just going by the content of that trace message.  I should have looked more closely.

**Comment by edreamleo** [comment]
@tbpassin I'm glad you *didn't* look more closely. The signature for the `goto.find_file_line` is confusing and redundant. I've just created PR #3187 to eliminate the confusion and simplify the code.

**Comment by edreamleo** [comment]
Fixed via PR #3215.
